### ISSUE TITLE
docker 1.12.1 go1.6.3 is unstable on macOS Sierra 10.12

### ISSUE THREAD

**Issue by achekulaev** [docker/docker#26798]
The latest [Docker release 1.12.1](https://get.docker.com/builds/Darwin/x86_64/docker-1.12.1.tgz) referenced in releases here on GitHub is compiled with go1.6.3 and is not stable on latest macOS sierra 10.12.

See https://github.com/docker/machine/issues/3773 for instability details I experienced as I initially thought it was a docker-machine issue but that was docker that caused it to crash.

Here is an example where I just repeatedly manually run `docker info >/dev/null`:

```
16:36:26 Alexeis-MacBook-Pro-2016 /Users/alexei.chekulaev/Projects/docksal/drupal8
$ eval $(docker-machine env docksal)
16:36:30 Alexeis-MacBook-Pro-2016 /Users/alexei.chekulaev/Projects/docksal/drupal8
$ docker info > /dev/null
16:36:35 Alexeis-MacBook-Pro-2016 /Users/alexei.chekulaev/Projects/docksal/drupal8
$ docker info > /dev/null
16:36:36 Alexeis-MacBook-Pro-2016 /Users/alexei.chekulaev/Projects/docksal/drupal8
$ docker info > /dev/null
16:36:37 Alexeis-MacBook-Pro-2016 /Users/alexei.chekulaev/Projects/docksal/drupal8
$ docker info > /dev/null
16:36:38 Alexeis-MacBook-Pro-2016 /Users/alexei.chekulaev/Projects/docksal/drupal8
$ docker info > /dev/null
16:36:38 Alexeis-MacBook-Pro-2016 /Users/alexei.chekulaev/Projects/docksal/drupal8
$ docker info > /dev/null
16:36:38 Alexeis-MacBook-Pro-2016 /Users/alexei.chekulaev/Projects/docksal/drupal8
$ docker info > /dev/null
16:36:38 Alexeis-MacBook-Pro-2016 /Users/alexei.chekulaev/Projects/docksal/drupal8
$ docker info > /dev/null
16:36:39 Alexeis-MacBook-Pro-2016 /Users/alexei.chekulaev/Projects/docksal/drupal8
$ docker info > /dev/null
16:36:39 Alexeis-MacBook-Pro-2016 /Users/alexei.chekulaev/Projects/docksal/drupal8
$ docker info > /dev/null
16:36:39 Alexeis-MacBook-Pro-2016 /Users/alexei.chekulaev/Projects/docksal/drupal8
$ docker info > /dev/null
16:36:39 Alexeis-MacBook-Pro-2016 /Users/alexei.chekulaev/Projects/docksal/drupal8
$ docker info > /dev/null
fatal error: unexpected signal during runtime execution
[signal 0xb code=0x1 addr=0x8e5f5906be0 pc=0x401357b]

runtime stack:
runtime.throw(0x49eb920, 0x2a)
    /usr/local/go/src/runtime/panic.go:547 +0x90
runtime.sigpanic()
    /usr/local/go/src/runtime/sigpanic_unix.go:12 +0x5a
runtime.unlock(0x4d1c780)
    /usr/local/go/src/runtime/lock_sema.go:107 +0x14b
runtime.(*mheap).alloc_m(0x4d1c780, 0x1, 0xd, 0x4f8f578)
    /usr/local/go/src/runtime/mheap.go:492 +0x314
...
```

At the same time [Docker for Mac](https://download.docker.com/mac/stable/Docker.dmg) installation that is distributed on docker.com contains different docker client version in `Docker.app/Contents/Resources/bin/docker`. That version is compiled with go1.7.1 and is working perfectly fine on macOS Sierra.

**Comment by justincormack** [comment]
Yes. The binaries used in Docker for Mac are compiled from https://github.com/justincormack/docker/tree/pinata-sierra-client

I believe we will release for 1.12.2 compiled like this too.

cc @icecrime @tiborvass

**Comment by thaJeztah** [comment]
ping @vieux as well - I'm not sure a decision was made on this (switching to a different Go version in a patch release)

**Comment by justincormack** [comment]
I think the consensus was that we should release a binary, even if the darwin build is different from the rest. I can do a patch for 1.12.2 branch that patches Go if absolutely necessary.

**Comment by thaJeztah** [comment]
Let me add it to the milestone at least, so that we don't forget to discuss

**Comment by thaJeztah** [comment]
Related; https://github.com/docker/docker/pull/26389

**Comment by jrslv** [comment]
All `docker-machine` and `docker` issues I experienced after upgrading to Sierra (go panics, `unexpected EOF` during `docker-machine create`, hanging `docker build`, segfaults and more) were fixed by this command: `sudo launchctl unload /System/Library/LaunchDaemons/org.ntp.ntpd.plist`

**Comment by justincormack** [comment]
PR for 1.12.2 here https://github.com/docker/docker/pull/26875

**Comment by gabrielqs** [comment]
What's the relation between the go version and NTPD here? Is there a suggested workaround while 1.12.2 doesn't come out? Thanks!

**Comment by justincormack** [comment]
There was a change to the OSX `gettimeofday` system call, so the issues are in time related code. The `ntpd` workaround only helps a little to alleviate the issue, it will not completely solve it. You can download Docker for Mac and copy the binary from there and use that it is in `/Applications/Docker.app/Contents/Resources/bin/docker`.

The 1.12.2-rc1 release candidate will be out on Tuesday.

**Comment by gabrielqs** [comment]
@justincormack thanks. I can confirm that using the docker binary distributed with the docker for mac package solved it for now.

**Comment by vieux** [comment]
fixed by #26875
